projects/components/src/load-async/loader/loader.component.ts

```diff
@@ -1,25 +1,47 @@
 import { ChangeDetectionStrategy, Component, Input, OnChanges } from '@angular/core';
 import { ImagesAssetPath } from '@hypertrace/assets-library';
+import { assertUnreachable } from '@hypertrace/common';
+import { SkeletonType } from '../../skeleton/skeleton.component';
 import { LoaderType } from '../load-async.service';
 
 @Component({
   selector: 'ht-loader',
   styleUrls: ['./loader.component.scss'],
   changeDetection: ChangeDetectionStrategy.OnPush,
   template: `
-    <div class="ht-loader">
-      <img [ngClass]="[this.currentLoaderType]" [src]="this.getImagePathFromType(this.currentLoaderType)" />
+    <div class="ht-loader" [ngClass]="{ 'flex-centered': this.isOldLoaderType }">
+      <ng-container *ngIf="!this.isOldLoaderType; else oldLoader">
+        <ht-skeleton [shapeStyle]="this.skeletonType" [repeat]="this.repeatLoaderCount"></ht-skeleton>
+      </ng-container>
+
+      <ng-template #oldLoader>
+        <img [ngClass]="[this.currentLoaderType]" [src]="this.getImagePathFromType(this.currentLoaderType)" />
+      </ng-template>
     </div>
   `
 })
 export class LoaderComponent implements OnChanges {
   @Input()
   public loaderType?: LoaderType;
 
+  @Input()
+  public repeatLoaderCount?: number;
+
+  public skeletonType: SkeletonType = SkeletonType.Rectangle;
+
   public currentLoaderType: LoaderType = LoaderType.Spinner;
 
+  public isOldLoaderType: boolean = true;
+
   public ngOnChanges(): void {
     this.currentLoaderType = this.loaderType ?? LoaderType.Spinner;
+
+    this.isOldLoaderType =
+      this.currentLoaderType === LoaderType.ExpandableRow ||
+      this.currentLoaderType === LoaderType.Spinner ||
+      this.currentLoaderType === LoaderType.Page;
+
+    this.skeletonType = this.getSkeletonTypeForLoader(this.currentLoaderType);
```

Comment:
we need these to be recomputed only if `loaderType` input has changed. Add a condition to check for that around these statements.